src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2294,6 +2286,10 @@ void MarkGenericParameter (GenericParameter parameter)
 			}
 		}
 
+		/// <summary>
+		/// Returns true if any of the base methods of the <paramref name="method"/> passed is in an assembly that is not trimmed (i.e. action != trim)
+		/// </summary>
+		/// <remarks>This ignores any base methods defined in interfaces. To also check methods defined in interfaces</remarks>
 		bool IsVirtualNeededByTypeDueToPreservedScope (MethodDefinition method)
 		{
 			if (!method.IsVirtual)
```

Comment:
I think with the current changes this will have to handle static interface methods as well.